<a href="https://colab.research.google.com/github/sisterme3/Data_Science/blob/master/NB2_SchoolProfile.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### NB2_SchoolProfile

The Essentials:
*   Upload the data

In [0]:
# Upload the data files from my github repository 
! git clone https://github.com/acreeg/dsc1010-datasets.git

Cloning into 'dsc1010-datasets'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 8 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (8/8), done.


*   Import necessary packages

In [0]:
#import pandas, the python package for data analytics
import pandas as pd

*   Create the dataframe

In [0]:
# Read the data_2015_school_profile.xlsx data file into a pandas dataframe
# Name the dataframe SchoolProfile 
filename = 'dsc1010-datasets/data_2015_school_profile.xlsx'
SchoolProfile = pd.read_excel(filename)

In [0]:
# Let's limit our exploration to High Schools
HSProfile = SchoolProfile[SchoolProfile['GRADES_SERVED'] == 'Grades 9-12']

What can we learn about HSProfile?
*   Column names

In [0]:
HSProfile.columns

Index(['YEAR', 'DISTRICT', 'DISTRICT_NAME', 'SCHOOL_NAME', 'SCHOOL_ID',
       'GRADES_SERVED', 'SAFE_SCHOOL', 'AVERAGE_DAILY_MEMBERSHIP', 'TOTAL',
       'WHITE', 'AFRICAN_AMERICAN', 'HISPANIC', 'ASIAN', 'NATIVE_AMERICAN',
       'HAWAIIAN_PACISLD', 'MALE', 'FEMALE', 'WHITE_PCT',
       'AFRICAN_AMERICAN_PCT', 'HISPANIC_PCT', 'ASIAN_PCT',
       'NATIVE_AMERICAN_PCT', 'HAWAIIAN_PACISLD_PCT', 'MALE_PCT', 'FEMALE_PCT',
       'WHITE_MALE', 'AFRICAN_AMERICAN_MALE', 'HISPANIC_MALE', 'ASIAN_MALE',
       'NATIVE_AMERICAN_MALE', 'HAWAIIAN_PACISLD_MALE', 'WHITE_FEMALE',
       'AFRICAN_AMERICAN_FEMALE', 'HISPANIC_FEMALE', 'ASIAN_FEMALE',
       'NATIVE_AMERICAN_FEMALE', 'HAWAIIAN_PACISLD_FEMALE',
       'LIMITED_ENGLISH_PROFICIENT_PCT', 'LIMITED_ENGLISH_PROFICIENT',
       'NUMBER_EXEMPT_FROM_READING_ASSESSMENT', 'STUDENTS_WITH_DISABILITIES',
       'STUDENTS_WITH_DISABILITIES_PCT', 'ECONOMICALLY_DISADVANTAGED',
       'ECONOMICALLY_DISADVANTAGED_PCT'],
      dtype='object')

In [0]:
# In order to select a single column from a dataframe
# ... we use the form: dataframe['column_name']
HSProfile['SCHOOL_NAME']

2                      Anderson County High School
3         Anderson County Career  Technical Center
8                              Clinton High School
27                           Oak Ridge High School
33                             Cascade High School
                           ...                    
1925                                   All Schools
1926                       Alvin C. York Institute
1932                Tennessee High School For Deaf
1951                           Fairley High School
1953    Martin Luther King Preparatory High School
Name: SCHOOL_NAME, Length: 317, dtype: object

In [0]:
# In order to select 2 or more columns,
# ... we need to create a list of column names
# For a list we use the following form:
Columns = ['YEAR', 'DISTRICT', 'DISTRICT_NAME', 'SCHOOL_NAME', 'SCHOOL_ID',
       'GRADES_SERVED', 'SAFE_SCHOOL', 'AVERAGE_DAILY_MEMBERSHIP', 'TOTAL',
       'WHITE', 'AFRICAN_AMERICAN', 'HISPANIC', 'ASIAN', 'NATIVE_AMERICAN',
       'HAWAIIAN_PACISLD', 'MALE', 'FEMALE', 'WHITE_PCT',
       'AFRICAN_AMERICAN_PCT', 'HISPANIC_PCT', 'ASIAN_PCT',
       'NATIVE_AMERICAN_PCT', 'HAWAIIAN_PACISLD_PCT', 'MALE_PCT', 'FEMALE_PCT',
       'WHITE_MALE', 'AFRICAN_AMERICAN_MALE', 'HISPANIC_MALE', 'ASIAN_MALE',
       'NATIVE_AMERICAN_MALE', 'HAWAIIAN_PACISLD_MALE', 'WHITE_FEMALE',
       'AFRICAN_AMERICAN_FEMALE', 'HISPANIC_FEMALE', 'ASIAN_FEMALE',
       'NATIVE_AMERICAN_FEMALE', 'HAWAIIAN_PACISLD_FEMALE',
       'LIMITED_ENGLISH_PROFICIENT_PCT', 'LIMITED_ENGLISH_PROFICIENT',
       'NUMBER_EXEMPT_FROM_READING_ASSESSMENT', 'STUDENTS_WITH_DISABILITIES',
       'STUDENTS_WITH_DISABILITIES_PCT', 'ECONOMICALLY_DISADVANTAGED',
       'ECONOMICALLY_DISADVANTAGED_PCT']

In [0]:
# Let's restrict our columns to only those we want to consider:
# Perhaps only 'DISTRICT_NAME', 'SCHOOL_NAME','MALE_PCT', 'FEMALE_PCT'
ColumnsChosen = ['DISTRICT_NAME', 'SCHOOL_NAME','MALE_PCT', 'FEMALE_PCT']

In [31]:
# Now let's consider HSProfile restricted to only these columns
HSProfile[['DISTRICT_NAME', 'SCHOOL_NAME','MALE_PCT', 'FEMALE_PCT']].head()

DISTRICT_NAME  ... FEMALE_PCT
2   Anderson County  ...  49.243856
3   Anderson County  ...        NaN
8   Anderson County  ...  47.798742
27        Oak Ridge  ...  49.388049
33   Bedford County  ...  44.465291

[5 rows x 4 columns]

In [30]:
# There is another way we can get the same results.
# Remember, we create a list of columns named ColumnsChosen
HSProfile[ColumnsChosen].head()

DISTRICT_NAME  ... FEMALE_PCT
2   Anderson County  ...  49.243856
3   Anderson County  ...        NaN
8   Anderson County  ...  47.798742
27        Oak Ridge  ...  49.388049
33   Bedford County  ...  44.465291

[5 rows x 4 columns]

Why do we put column names in quotes, like ... 'GRADES_SERVED', but not ColumnsChosen?

What do you think?

ColumnsChosen are an array type which means that it can be called by its name. While 'GRADES_SERVED' is apart of the dataset columns so it must be called in quotes.

*   Descriptive Statistics

In [0]:
# Suppose we're interested in the percentage of females across all Tennessee High Schools
# What do you suppose the highest percentage might be?
HSProfile['FEMALE_PCT'].max()

# the lowest?
#HSProfile['FEMALE_PCT'].min()

68.1318681318681

In [0]:
# Each cell only prints out the last output from a cell.
# If you want more, you can use the print command.
print('Female PCT')
print('Max:') 
print(HSProfile['FEMALE_PCT'].max())
print()
print('Min:') 
print(HSProfile['FEMALE_PCT'].min())


Female PCT
Max:
68.1318681318681

Min:
21.9512195121951


In [0]:
# We may want to know even more information.
# Perhaps we would like to know about the distribution of 'FEMALE_PCT'
# What we want is a look at the Descriptive Statistics.
HSProfile['FEMALE_PCT'].describe()

count    288.000000
mean      48.609365
std        5.027362
min       21.951220
25%       47.215994
50%       48.783867
75%       50.109844
max       68.131868
Name: FEMALE_PCT, dtype: float64

What do the descriptive statistics tell us about the distribution of values in the case of FEMALE_PCT across Tennessee High Schools?

What do you think?

The descriptive statistics tell us that across Tennessee High Schools, there are 288 female students across Tennessee High Schools and the average amount of females in each high school is 48.6%. Also the smallest amount of females in a high school is 21.9% and the max amount in high school is 68.13%.

In [0]:
# How would you locate the schools with the greatest/smallest FEMALE_PCT?
FemaleMax = HSProfile['FEMALE_PCT'].max()
FemaleMin = HSProfile['FEMALE_PCT'].min()

# Show all schools where the max (min) is achieved
FMaxSchools = HSProfile[HSProfile['FEMALE_PCT']== FemaleMax]
FMaxSchools

YEAR  DISTRICT  ... ECONOMICALLY_DISADVANTAGED ECONOMICALLY_DISADVANTAGED_PCT
301  2015       190  ...                      106.0                      58.241758

[1 rows x 44 columns]